firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/NativeSessionFileGzipper.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.common;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.zip.GZIPOutputStream;
+
+/** Copies {@link NativeSessionFile} into gzipped files within a provided path. */
+class NativeSessionFileGzipper {
+
+  static void processNativeSessions(File nativeSessionDirectory, List<NativeSessionFile> streams)
+      throws IOException {
+
+    for (NativeSessionFile stream : streams) {
+      InputStream inputStream = null;
+      try {
+        inputStream = stream.getStream();
+        if (inputStream == null) {
+          continue;
+        }
+        gzipInputStream(inputStream, new File(nativeSessionDirectory, stream.getName()));
+      } catch (Exception e) {
+        // Skip invalid files.
+      } finally {
+        CommonUtils.closeQuietly(inputStream);
+      }
+    }
+  }
+
+  private static void gzipInputStream(@Nullable InputStream input, @NonNull File output)
+      throws IOException {
+    if (input == null) {
+      return;
+    }
+    byte[] buffer = new byte[1024];
```

Comment:
We should probably use 8192 here, as above.

Reply:
This was copypasta of the other place where we were zipping, happy to switch it (I assumed each was set for domain-specific reasons of the objects they were serializing)

Reply:
Understood, yeah, no good reason for it.